Themes/default/scripts/jquery.sceditor.smf.js

```diff
@@ -734,63 +733,61 @@ sceditor.formats.bbcode.set(
-	'url', {
+	'email', {
 		allowsEmpty: true,
-		quoteType: $.sceditor.BBCodeParser.QuoteType.never,
+		quoteType: sceditor.BBCodeParser.QuoteType.never,
 		tags: {
 			a: {
-				href: null
+				'data-type': ['email']
 			}
 		},
-		format: function (element, content) {
-			var element = $(element),
-				url = element.attr('href');
-
-			// make sure this link is not an e-mail, if it is return e-mail BBCode
-			if (url.substr(0, 7) === 'mailto:')
-				return '[email=' + url.substr(7) + ']' + content + '[/email]';
-
-			if (typeof element.attr('target') !== "undefined")
-				return '[url=\"' + decodeURI(url) + '\"]' + content + '[/url]';
-
-			// A mention?
-			else if (typeof element.attr('data-mention') !== "undefined")
-				return '[member='+ element.attr('data-mention') +']'+ content.replace('@','') +'[/member]';
-
-			// Is this an attachment?
-			else if (typeof element.attr('data-attachment') !== "undefined")
-			{
-				var attribs = ' id=' + element.attr('data-attachment');
-				if (typeof element.attr('alt') !== "undefined")
-					attribs += ' alt=' + element.attr('alt');
-				if (typeof element.attr('data-type') !== "undefined")
-					attribs += ' type=' + element.attr("data-type");
+		format: function (element, content)
+		{
+			return '[email=' + decodeURI(element.href) + ']' + content + '[/email]';
+		},
+		html: function (token, attrs, content)
+		{
+			return '<a data-type="email" href="' + encodeURI(attrs.defaultattr || content) + '">' + content + '</a>';
+		}
+	}
+);
 
-				return '[attach' + attribs + ']' + content + '[/attach]';
+sceditor.formats.bbcode.set(
+	'url', {
+		allowsEmpty: true,
+		quoteType: sceditor.BBCodeParser.QuoteType.never,
+		tags: {
+			a: {
+				'data-type': ['url']
 			}
-
-			else
-				return '[iurl=\"' + decodeURI(url) + '\"]' + content + '[/iurl]';
 		},
-		html: function (token, attrs, content) {
-			if (typeof attrs.defaultattr === "undefined" || attrs.defaultattr.length === 0)
-				attrs.defaultattr = content;
-
-			return '<a target="_blank" rel="noopener" href="' + encodeURI(attrs.defaultattr) + '">' + content + '</a>';
+		format: function (element, content)
+		{
+			return '[url=' + decodeURI(element.href) + ']' + content + '[/url]';
+		},
+		html: function (token, attrs, content)
+		{
+			return '<a data-type="url" href="' + encodeURI(attrs.defaultattr || content) + '">' + content + '</a>';
 		}
 	}
 );
 
 sceditor.formats.bbcode.set(
 	'iurl', {
 		allowsEmpty: true,
-		quoteType: $.sceditor.BBCodeParser.QuoteType.never,
-		html: function (token, attrs, content) {
-
-			if (typeof attrs.defaultattr === "undefined" || attrs.defaultattr.length === 0)
-				attrs.defaultattr = content;
-
-			return '<a href="' + encodeURI(attrs.defaultattr) + '">' + content + '</a>';
+		quoteType: sceditor.BBCodeParser.QuoteType.never,
+		tags: {
+			a: {
+				'data-type': ['iurl']
+			}
+		},
+		format: function (element, content)
+		{
+			return '[iurl=' + decodeURI(element.href) + ']' + content + '[/iurl]';
```

Comment:
```suggestion
			return '[iurl=\"' + decodeURI(element.href) + '\"]' + content + '[/iurl]';
```

Please keep the quotation marks. See #3565.